index.d.ts

```diff
@@ -0,0 +1,1353 @@
+     * entry which accepts `transactionId` allows to call `cancelTransaction` function. When cancelled operation is a
+     * promise or a callback which registers errors, {@link #bleerror|BleError} with error code
+     * {@link #bleerrorcodeoperationcancelled|OperationCancelled} will be emitted in that case. Cancelling transaction
+     * which doesn't exist is ignored.
+     *
+     * @example
+     * const transactionId = 'monitor_battery';
+     *
+     * // Monitor battery notifications
+     * manager.monitorCharacteristicForDevice(
+     *   device.id, '180F', '2A19',
+     *   (error, characteristic) => {
+     *   // Handle battery level changes...
+     * }, transactionId);
+     *
+     * // Cancel after specified amount of time
+     * setTimeout(() => manager.cancelTransaction(transactionId), 2000);
+     *
+     * @param {string} transactionId Id of pending transactions.
+     */
+    cancelTransaction(transactionId: string): void;
+
+    // Mark: Monitoring state --------------------------------------------------------------------------------------------
+
+    /**
+     * Enable Bluetooth. This function blocks until BLE is in PoweredOn state. [Android only]
+     *
+     * @param {?string} transactionId Transaction handle used to cancel operation
+     * @returns {Promise<BleManager>} Promise completes when state transition was successful.
+     */
+    enable(transactionId?: string): Promise<BleManager>;
+
+    /**
+     * Disable Bluetooth. This function blocks until BLE is in PoweredOff state. [Android only]
+     *
+     * @param {?string} transactionId Transaction handle used to cancel operation
+     * @returns {Promise<BleManager>} Promise completes when state transition was successful.
+     */
+    disable(transactionId?: string): Promise<BleManager>;
+
+    /**
+     * Current, global {@link State} of a {@link BleManager}. All APIs are working only when active state
+     * is "PoweredOn".
+     *
+     * @returns {Promise<State>} Promise which emits current state of BleManager.
+     */
+    state(): Promise<State> {
+      return this._callPromise(BleModule.state());
+    }
+
+    /**
+     * Notifies about {@link State} changes of a {@link BleManager}.
+     *
+     * @example
+     * const subscription = this.manager.onStateChange((state) => {
+     *      if (state === 'PoweredOn') {
+     *          this.scanAndConnect();
+     *          subscription.remove();
+     *      }
+     *  }, true);
+     *
+     * @param {function(newState: State)} listener Callback which emits state changes of BLE Manager.
+     * Look at {@link State} for possible values.
+     * @param {boolean} [emitCurrentState=false] If true, current state will be emitted as well. Defaults to false.
+     *
+     * @returns {Subscription} Subscription on which `remove()` function can be called to unsubscribe.
+     */
+    onStateChange(listener: (newState: State) => void, emitCurrentState: boolean = false): Subscription;
```

Comment:
`onStateChange(listener: (newState: State) => void, emitCurrentState?: boolean): Subscription;`
Initializer is not allowed.